apps/backend/src/auth/ProposalAuthorization.ts

```diff
@@ -224,20 +192,62 @@ export class ProposalAuthorization {
         ).some((value) => value)
       : false;
 
-    return (
-      this.userAuth.isUserOfficer(agent) ||
-      this.userAuth.isSampleSafetyReviewer(agent) ||
-      this.userAuth.isInstrumentScientist(agent) ||
-      this.userAuth.hasGetAccessByToken(agent) ||
-      (await this.isMemberOfProposal(agent, proposal)) ||
-      (await this.isReviewerOfProposal(agent, proposal.primaryKey)) ||
-      (await this.isScientistToProposal(agent, proposal.primaryKey)) ||
-      (await this.isInstrumentManagerToProposal(agent, proposal.primaryKey)) ||
-      isInternalReviewerOnSomeTechnicalReview ||
-      (await this.isChairOrSecretaryOfProposal(agent, proposal.primaryKey)) ||
-      (await this.isVisitorOfProposal(agent, proposal.primaryKey)) ||
-      (await this.isMemberOfFapProposal(agent, proposal.primaryKey))
-    );
+    return isInternalReviewerOnSomeTechnicalReview;
+  }
+
+  async hasReadRights(
+    agent: UserWithRole | null,
+    proposalOrProposalId: Proposal | number
+  ): Promise<boolean> {
+    if (!agent) {
+      return false;
+    }
+
+    const proposal = await this.resolveProposal(proposalOrProposalId);
+
+    if (!proposal) {
+      return false;
+    }
+
+    const currentRole = agent?.currentRole?.shortCode;
+
+    let haveAccess = false;
+
+    switch (currentRole) {
+      case Roles.USER:
+        haveAccess =
+          (await this.isMemberOfProposal(agent, proposal)) ||
+          (await this.isVisitorOfProposal(agent, proposal.primaryKey));
+        break;
+      case Roles.INSTRUMENT_SCIENTIST:
+        haveAccess =
+          (await this.isInstrumentManagerToProposal(
+            agent,
+            proposal.primaryKey
+          )) || (await this.isScientistToProposal(agent, proposal.primaryKey));
+        break;
+      case Roles.INTERNAL_REVIEWER:
+        haveAccess = await this.isInternalReviewer(agent, proposal.primaryKey);
+        break;
+      case Roles.FAP_REVIEWER:
+      case Roles.FAP_SECRETARY:
+      case Roles.FAP_CHAIR:
+        haveAccess = await this.isMemberOfFapProposal(
+          agent,
+          proposal.primaryKey
+        );
+        break;
+      case Roles.USER_OFFICER:
+        haveAccess = this.userAuth.isUserOfficer(agent);
```

Comment:
```suggestion
        haveAccess = true;
```
I think the test case needs to be updated